README.md

```diff
@@ -295,16 +398,380 @@ Here is an example:
+               :initform ()
+               :type (list-of symbol))))
+```
+``` emacs-lisp
+(jeison-defclass jeison-job nil
+  ((company :initarg :company)
+   (position :initarg :position :path title)
+   (location :initarg :location :path (location city))))
+```
+``` emacs-lisp
+(jeison-defclass jeison-skill nil
+  ((name :initarg :name :type string)
+   (description :initarg :level :type string)))
+```
+
+Note 1: The `:initform` value in a `jeison-hash-table-of`-typed slot
+declaration is not particularly important because *jeison* intializes
+and sets the object's slot values before EIEIO checks the newly
+created object's type constraints. **Do not** use `nil` as the
+initializer because this will cause an invalid type failure when
+`jeison-defclass` defines the class and EIEIO type checks the
+`:initform` value.
+
+Note 2: This part of the example uses symbols as the hash key
+type. You can also use strings, which is discussed further on. The
+`"skill-1"`, `"skill-2"` and `"skill-3"` object member strings in the
+`skills` JSON object are converted to intern-ed symbols, i.e.,
+`'skill-1`, `'skill-2` and `'skill-3`. This means that you need to use
+`list-of symbol` to ensure that the skill reference strings in the
+`skill-refs` arrays are also converted into symbols to avoid a type
+mismatch when looking up elements with `gethash`.
+
+Using the following JSON object as an Emacs Lisp string:
+
+``` emacs-lisp
+(setq json-people "{
+  \"people\": [
+    {
+      \"personal\": {
+        \"address\": { },
+        \"name\": {
+          \"first\": \"John\",
+          \"last\": \"Johnson\"
+        }
+      },
+      \"job\": {
+        \"company\": \"Good Inc.\",
+        \"title\": \"CEO\",
+        \"location\": {
+          \"country\": \"Norway\",
+          \"city\": \"Oslo\"
+        }
+      },
+      \"skill-refs\": [
+        \"skill-1\",
+        \"skill-3\"
+      ]
+    },
+    {
+      \"personal\": {
+        \"address\": { },
+        \"name\": {
+          \"first\": \"Sammy\",
+          \"last\": \"Smith\"
+        }
+      },
+      \"job\": {
+        \"company\": \"Good Inc.\",
+        \"title\": \"VP, Marketing\",
+        \"location\": {
+          \"country\": \"USA\",
+          \"city\": \"Peoria\"
+        }
+      },
+      \"skill-refs\": [
+        \"skill-2\",
+        \"skill-3\"
+      ]
+    }
+  ],
+  \"skills\": {
+    \"skill-1\": {
+      \"name\": \"Programming\",
+      \"description\": \"Can code like Richard S or Linus T\"
+    },
+    \"skill-2\": {
+      \"name\": \"Design\",
+      \"description\": \"Can think out loud, structure code systems\"
+    },
+    \"skill-3\": {
+      \"name\": \"Communication\",
+      \"description\": \"Can explain complicated things to social scientists\"
+    }
+  }
+}")
+```
+
+The code below demonstrates how to use the `skills` hash table and
+symbols as keys:
+
+```emacs-lisp
+(let* ((people-data (jeison-read jeison-people json-people)))
```

Comment:
Can you please simplify this code to be less nested and to match the style of other examples? 
I suggest to add `setq`s for `people-data`, `people` and `skills`, and remove explicit prints.